### PR TITLE
Feature/add change return types

### PR DESCRIPTION
**Pull request by martinyde**
Change return types, and add a few.

I'm unsure if $expiry; in CacheItem.php should be made public or removed. It's  never used within the class but all that the two expire methods do is set it. So what gives???

### CODE REVIEW

File: drupal_psr6_cache.module [hunk 1]
```diff
@@ -10,15 +10,18 @@ use Drupal\Core\Routing\RouteMatchInterface;
...
  */
-function drupal_psr6_cache_help($route_name, RouteMatchInterface $route_match) {
+function drupal_psr6_cache_help(string $route_name, RouteMatchInterface $route_match): string {
+  $output = '';
+
   switch ($route_name) {
     // Main module help for the drupal_psr6_cache module.
     case 'help.page.drupal_psr6_cache':
-      $output = '';
+
```
Comment (rimi-itk) [inline drupal_psr6_cache.module:18]: Remove this empty line:
```suggestion
```

File: src/Cache/CacheItem.php [hunk 2]
```diff
@@ -85,18 +85,12 @@ public function set($value): self {
...
     }
-    elseif ($expiration instanceof DateTimeInterface) {
-      $this->expiry = (float) $expiration->format('U.u');
-    }
     else {
-      throw new \RuntimeException(sprintf(
-        'Expected $expiration to be an instance of DateTimeInterface or null, got %s',
-        is_object($expiration) ? get_class($expiration) : gettype($expiration)
-      ));
+      $this->expiry = (float) $expiration->format('U.u');
```
Comment (rimi-itk) [inline src/Cache/CacheItem.php:93]: Why has the `instanceof` check been removed?

File: src/Cache/CacheItem.php [hunk 3]
```diff
@@ -105,21 +99,15 @@ public function expiresAt($expiration): self {
...
-      $this->expiry = $time + microtime(TRUE);
+    elseif ($time instanceof \DateInterval) {
+      $this->expiry = microtime(TRUE) + (float) \DateTime::createFromFormat('U', '0')->add($time)->format('U.u');
     }
     else {
-      throw new \RuntimeException(sprintf(
-        'Expected $time to be either an integer, an instance of DateInterval or null, got %s',
-        is_object($time) ? get_class($time) : gettype($time)
-      ));
+      $this->expiry = $time + microtime(TRUE);
```
Comment (rimi-itk) [inline src/Cache/CacheItem.php:110]: Why has the type check (`is_int`) been removed?

**Review by rimi-itk — CHANGES_REQUESTED** [review]

**Comment by rimi-itk** [comment]
I would leave `$expiry` as it is, i.e. `private`, but it sure seems weird to have a private property that's not used.

File: src/Cache/CacheItem.php (hunk 2 above)
Comment (martinyde) [reply src/Cache/CacheItem.php:93]: According to the interface definition  $expiration can only be DateTimeInterface|null.
https://github.com/php-fig/cache/blob/3.0.0/src/CacheItemInterface.php and we are checking for the null case.

File: src/Cache/CacheItem.php (hunk 3 above)
Comment (martinyde) [reply src/Cache/CacheItem.php:110]: According to the interface definition $time can only be DateInterval|int|null.
https://github.com/php-fig/cache/blob/3.0.0/src/CacheItemInterface.php and we are checking for the other two.

**Pull request merged**